# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: toolkit/components/url-classifier/LookupCacheV4.cpp
# Commit: c3baab96e392
# Full Hash: c3baab96e39286879ea0eae1a74631c103fcc8cb
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2025-06-19 21:40:31
# Description:
#   Bug 1972147 - Reduce stack allocation in LookupCacheV4::ApplyUpdate. r=dimi
#   
#   Seeing that bug 1968812 by adding 16 bytes to the stack in TaskQueue::Runner
#   (on 64-bit systems) caused a stack buffer overrun, this reduction of 32 bytes in
#   the crashing stack should be sufficient compensation.
# ==============================================================================

diff -r 089becdb709e -r c3baab96e392 toolkit/components/url-classifier/LookupCacheV4.cpp
--- a/toolkit/components/url-classifier/LookupCacheV4.cpp	Thu Jun 19 08:18:30 2025 +0000
+++ b/toolkit/components/url-classifier/LookupCacheV4.cpp	Thu Jun 19 08:18:30 2025 +0000
@@ -425,7 +425,7 @@
   // after merging the data with local prefixes.
   aTableUpdate->Clear();
 
-  nsAutoCString sha256;
+  nsAutoCStringN<40> sha256;
   crypto->Finish(false, sha256);
   if (aTableUpdate->SHA256().IsEmpty()) {
     LOG(("Update sha256 hash missing."));
